### PR TITLE
improve: domxss,sqliplugin, ascanrules: Add/update Top10 Mappings

### PR DESCRIPTION
**Pull request by kingthorin**
ascanrules
- Update XSS rules to use the XSS mapping for 2017.

domxss & sqliplugin
- Updated scan rules and unit tests (some unit test classes added).
- Fixed naming (and help link) for MySql scan rule and unit test.
- Update build files to depend on commonlib.
- Added note to CHANGELOG.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Review by psiinon — APPROVED** [review]

**Comment by thc202** [comment]
The changelog of DOM XSS was not updated.

**Comment by kingthorin** [comment]
accessControl still needs to be tackled I guess. I guess I'll need to set it to target 2.11 so that I can specifically set the tags in the raise alert methods (authN and authZ are handled separately). Should I leave that for another PR? Is accessControl covered on the alert pages?

File: addOns/domxss/domxss.gradle.kts [hunk 1]
```diff
@@ -21,13 +21,19 @@ zapAddOn {
...
             }
         }
     }
 }
 
 dependencies {
+    compileOnly(parent!!.childProjects.get("commonlib")!!)
     compileOnly(parent!!.childProjects.get("selenium")!!)
+    testImplementation(parent!!.childProjects.get("commonlib")!!)
+    testImplementation(parent!!.childProjects.get("commonlib")!!.sourceSets.test.get().output)
```
Comment (thc202) [inline addOns/domxss/domxss.gradle.kts:36]: This is not really needed (same for the other change).

**Comment by thc202** [comment]
I'm updating all add-ons to 2.11, will PR shortly.

**Comment by thc202** [comment]
The commit/PR message needs tweaks (e.g. test class, MySql reference).

**Comment by kingthorin** [comment]
Oops thanks

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**